service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -41,182 +41,244 @@
-            return SERVICE_REQUESTED + extractServiceRequested(referralRequest);
+        private List<String> retrieveDescription() {
+            return List.of(
+                buildIdentifierDescription(),
+                buildPriorityDescription(),
+                buildServiceRequestedDescription(),
+                buildSpecialtyDescription(),
+                buildRecipientDescription(),
+                buildReasonCodeDescription(),
+                buildNoteDescription(),
+                buildTextDescription()
+            );
         }
-        return StringUtils.EMPTY;
-    }
 
-    private String buildRequesterDescription(ReferralRequest referralRequest) {
-        if (referralRequest.hasRequester() && referralRequest.getRequester().getAgent().hasReference()) {
-            IIdType reference = referralRequest.getRequester().getAgent().getReferenceElement();
-            if (reference.getResourceType().equals(ResourceType.Device.name())) {
-                return messageContext.getInputBundleHolder()
-                    .getResource(reference)
-                    .map(resource -> (Device) resource)
-                    .map(value -> extractDevice(value))
-                    .map(text -> REQUESTER_DEVICE + text)
-                    .orElseThrow(() -> new EhrMapperException("Could not resolve Device Reference"));
-            } else if (reference.getResourceType().equals(ResourceType.Organization.name())) {
-                return messageContext.getInputBundleHolder()
-                    .getResource(reference)
-                    .map(resource -> (Organization) resource)
-                    .map(value -> REQUESTER_ORG + value.getName())
-                    .orElseThrow(() -> new EhrMapperException("Could not resolve Organization Reference"));
-            } else if (reference.getResourceType().equals(ResourceType.Patient.name())) {
-                return REQUESTER_PATIENT;
-            } else if (reference.getResourceType().equals(ResourceType.RelatedPerson.name())) {
-                return messageContext.getInputBundleHolder()
-                    .getResource(reference)
-                    .map(resource -> (RelatedPerson) resource)
-                    .map(value -> REQUESTER_RELATION + extractHumanName(value.getNameFirstRep()))
-                    .orElseThrow(() -> new EhrMapperException("Could not resolve RelatedPerson Reference"));
-            } else if (reference.getResourceType().equals(ResourceType.Practitioner.name())) {
-                return StringUtils.EMPTY;
+        private String buildIdentifierDescription() {
+            return Optional.of(referralRequest).stream()
+                .filter(ReferralRequest::hasIdentifier)
+                .map(ReferralRequest::getIdentifier)
+                .flatMap(List::stream)
+                .filter(Objects::nonNull)
+                .filter(val -> UBRN_SYSTEM_URL.equals(val.getSystem()))
+                .map(Identifier::getValue)
+                .findAny()
```

Comment:
We were finding the first element previously, does `.findAny()` change anything?

Reply:
In most cases they're equivalent, pragmatically; for this code, we're looking at the values of identifiers with the UBRN system which I'd expect there only to be one of in good data. In dodgy data with multiples... they _should_ have the same value, right? These things are meant to identify the thing they represent after all. If so, findAny or findFirst are equivalent, and if not then... GIGO?

Reply:
Ok, fair enough 👍